twine/commands/upload.py

```diff
@@ -48,6 +50,47 @@ def skip_upload(response, skip_existing, package):
             (response.status_code == 403 and msg_403 in response.text)))
 
 
+def check_status_code(response, verbose):
+    """
+    Additional safety net to catch response code 410 in case the
+    UploadToDeprecatedPyPIDetected exception breaks.
+    Also includes a check for response code 405 and prints helpful error
+    message guiding users to the right repository endpoints.
+    """
+    if (response.status_code == 410 and
+            response.url.startswith(("https://pypi.python.org",
+                                     "https://testpypi.python.org"))):
+        raise exceptions.\
```

Comment:
I think the preferred formatting would be:

```python
raise exceptions.UploadToDeprecatedPyPIDetected(
    f"It appears you're uploading to pypi.python.org (or testpypi.python.org)."
    f"You've received a 410 error response."
    # ...
)
```

That said, I think it'd be more readable/maintainable to define both of these error messages in their respective exception classes. We could also modify `UploadToDeprecatedPyPIDetected` so that it doesn't need arguments, but changing that would require tweaking the other code that uses it.

Reply:
How do you feel about raising a single exception for `InvalidPyPIUploadURL` and `UploadToDeprecatedPyPIDetected` as they're both in the realm of invalid URLs, and passing in relevant messages when raising them?

Reply:
Fixed formatting.